rholang/src/main/scala/coop/rchain/rholang/interpreter/compiler/normalize.scala

```diff
@@ -1072,6 +1074,201 @@ object ProcNormalizeMatcher {
                 }
         } yield res
 
+      case p: PLet => {
+
+        p.decls_ match {
+
+          case concDeclsImpl: ConcDeclsImpl =>
+            def extractNamesAndProcs(decl: Decl): (ListName, NameRemainder, ListProc) =
+              decl match {
+                case declImpl: DeclImpl =>
+                  (declImpl.listname_, declImpl.nameremainder_, declImpl.listproc_)
+              }
+
+            val (listNames, listNameRemainders, listProcs) =
+              (extractNamesAndProcs(p.decl_) :: concDeclsImpl.listconcdecl_.toList.map {
+                case concDeclImpl: ConcDeclImpl => extractNamesAndProcs(concDeclImpl.decl_)
+              }).unzip3
+
+            /*
+             It is not necessary to use UUIDs to achieve concurrent let declarations.
+             While there is the possibility for collisions with either variables declared by the user
+             or variables declared within this translation, the chances for collision are astronomically
+             small (see analysis here: https://towardsdatascience.com/are-uuids-really-unique-57eb80fc2a87).
+             A strictly correct approach would be one that performs a ADT rather than an AST translation, which
+             was not done here due to time constraints.
+             */
+            val variableNames = List.fill(listNames.size)(UUID.randomUUID().toString)
+
+            val psends = variableNames.zip(listProcs).map {
+              case (variableName, listProc) =>
+                new PSend(new NameVar(variableName), new SendSingle(), listProc)
+            }
+
+            val pinput = {
+              val listLinearBind = new ListLinearBind()
+              for (linearBind <- variableNames.zip(listNames).zip(listNameRemainders).map {
+                                  case ((variableName, listName), nameRemainder) =>
+                                    new LinearBindImpl(
+                                      listName,
+                                      nameRemainder,
+                                      new NameVar(variableName)
+                                    )
+                                })
+                listLinearBind.add(linearBind)
```

Comment:
`.foreach(listLinearBind.add)` can do the trick here. On two other places can also be one liner.